### PR TITLE
add basic cypress workflow

### PR DESCRIPTION
**Pull request by ahmadabdolsaheb**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)

Closes #XXXXX

### CODE REVIEW

File: .github/workflows/cypress [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Cypress Tests
```
Comment (huyenltnguyen) [inline .github/workflows/cypress:1]: According to [the doc](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#about-yaml-syntax-for-workflows), I think we need to make the file a .yaml one.

**Review by raisedadead — COMMENTED** [review]
If we are adding a template let's put in some more efforts and model something on the main repo's template instead with at least one meanigful test.

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Cypress Tests
+
+on: [push, pull_request]
+
+jobs:
+  cypress-run:
+    name: Test
+    runs-on: ubuntu-18.04
```
Comment (ojeytonwilliams) [inline .github/workflows/cypress.yml:8]:
```suggestion
    # matching current Netlify build environment
    runs-on: ubuntu-16.04
```
Netlify builds in 16.04, so I think it makes sense to match that.

File: .github/workflows/cypress.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    strategy:
+      matrix:
+        browsers: [chrome, firefox]
+        node-version: [14.x]
```
Comment (ojeytonwilliams) [inline .github/workflows/cypress.yml:12]: Can we pin this in the package.json via the `engines` property?

```
  "engines": {
    "node": "^14.0.0",
    "npm": "^6.14.12"
  }
```
for example.

I went ahead and set  `NODE_VERSION=14` in Netlify, so the next build should use it.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**